pkg/util/gkenetworkparamsetutil/controller_utils.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gkenetworkparamsetutil
+
+import (
+	"fmt"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	networkv1alpha1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
+)
+
+// CreateAddGKENetworkParamSetHandler creates an add GKENetworkParamSet handler.
+func CreateAddGKENetworkParamSetHandler(f func(params *networkv1alpha1.GKENetworkParamSet) error) func(obj interface{}) {
+	return func(originalObj interface{}) {
+		params := originalObj.(*networkv1alpha1.GKENetworkParamSet).DeepCopy()
+		if err := f(params); err != nil {
+			utilruntime.HandleError(fmt.Errorf("Error while processing GkeNetworkParamSet Add: %v", err))
+		}
+	}
+}
+
+// CreateUpdateGKENetworkParamSetHandler creates an update GKENetworkParamSet handler.
+func CreateUpdateGKENetworkParamSetHandler(f func(oldParams, newParams *networkv1alpha1.GKENetworkParamSet) error) func(oldObj, newObj interface{}) {
+	return func(origOldObj, origNewObj interface{}) {
+		params := origNewObj.(*networkv1alpha1.GKENetworkParamSet).DeepCopy()
+		prevParams := origOldObj.(*networkv1alpha1.GKENetworkParamSet).DeepCopy()
+
+		if err := f(prevParams, params); err != nil {
+			utilruntime.HandleError(fmt.Errorf("Error while processing GkeNetworkParamSet Update: %v", err))
+		}
+	}
```

Comment:
no. need to add new packages if this is only imported in one place, can you please add the code directly in the file of the controller as a private function?